scripts/fix_unclaimed_records.py

```diff
@@ -0,0 +1,33 @@
+import sys
+import logging
+from website.app import setup_django
+
+setup_django()
+from osf.models import OSFUser
+logger = logging.getLogger(__name__)
+
+
+def add_missing_unclaimed_record(user, node, dry_run):
+    # Get referrer from logs
+    for log in node.logs.filter(action='contributor_added').order_by('date'):
+        if user._id in log.params['contributors']:
+            if log.user_id:
+                referrer = OSFUser.objects.get(id=log.user_id)
+                user.add_unclaimed_record(node, referrer, user.fullname)
+                if not dry_run:
+                    user.save()
+                logger.info('User {} has been given an unclaimed record with name {}'.format(user._id, user.fullname))
+            else:
+                logger.info('User {} has could not be given a record, because their refferer was anonymous'.format(user._id))
+
+
+def main(dry_run=True):
+    users = OSFUser.objects.filter(unclaimed_records={}, is_registered=False).filter(nodes__type='osf.node').include(None).distinct()
```

Comment:
Two issues here:
1. Pretty sure this query will try to add unclaimed records for disabled accounts.
2. This isn't going to catch unclaimed users who were both added to a node and via the preprints app to a different node.

Reply:
1. I don't see why we wounldn't add records for deactivated accounts, if the accounts get reactivated it's just another thing we have to remember to do.
2. gotcha 👍

Reply:
My understanding is that unclaimed records exist so that we can display usernames for unregistered contributors as an admin on a project added them. When that user registers the unclaimed records are not used anymore (I believe they are deleted now). The user specifies what their name is and how they want to be displayed so we don't need to infer from an admin's input of an unregistered user anymore.

I don't think it is possible to go from unclaimed -> disabled without being activated in the interim (and thus removing the need for unclaimed records). As such, I'm not sure adding unclaimed records for disabled users is a desirable outcome.